lib/symlink/prepare.js

```diff
@@ -24,12 +23,17 @@ function prepareSymlink(folderResolver, optResolver) {
     var writePath = path.resolve(basePath, file.relative);
 
     // Wire up new properties
+    // TODO new stats will have invalid props, but target stats will have the
+    // wrong ones...
     file.stat = (file.stat || new fs.Stats());
-    file.stat.mode = mode;
+    file.stat.mode = LINK_MODE;
     file.cwd = cwd;
     file.base = basePath;
     file.symlink = file.path;
     file.path = writePath;
+    // We have to set contents to null for a link
+    // Otherwise `isSymbolic()` returns false
+    file.contents = null;
```

Comment:
@erikkemperman this was an interesting find.  It makes sense, but I forgot we checked `isNull()` on the Vinyl